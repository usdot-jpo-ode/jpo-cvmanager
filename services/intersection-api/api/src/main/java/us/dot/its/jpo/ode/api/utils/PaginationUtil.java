package us.dot.its.jpo.ode.api.utils;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class PaginationUtil {

    public static <T> ResponseEntity<List<T>> createResponseEntityWithPaginationHeaders(Page<T> page) {
        HttpHeaders headers = new HttpHeaders();
        headers.add("X-Total-Count", String.valueOf(page.getTotalElements()));
        headers.add("X-Total-Pages", String.valueOf(page.getTotalPages()));
        headers.add("X-Has-More", String.valueOf(page.hasNext()));
        return new ResponseEntity<List<T>>(page.getContent(), headers, HttpStatus.OK);
    }
}