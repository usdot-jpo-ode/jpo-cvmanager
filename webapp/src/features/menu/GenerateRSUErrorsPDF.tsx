import React, { useRef, useState } from 'react'
import { useReactToPrint } from 'react-to-print'
import Button from '@mui/material/Button'
import { StyledEngineProvider, ThemeProvider } from '@mui/material'
import { theme } from '../../styles'
import AdminTable from '../../components/AdminTable'
import MaterialTable from '@material-table/core'

const GenerateRSUErrorsPDF = (props) => {
  const contentRef = useRef(null)
  const errorRef = useRef(null)
  const handlePrint = useReactToPrint({ contentRef })
  const handleErrorPrint = useReactToPrint({ contentRef: errorRef })

  return (
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <div className={'containerDiv'}>
          <div style={{ display: 'flex', justifyContent: 'space-between', marginBottom: '1rem' }}>
            <Button
              style={{ color: 'white' }}
              onClick={() => {
                handlePrint()
              }}
            >
              Print Full Report
            </Button>
            <Button
              style={{ color: 'white' }}
              onClick={() => {
                handleErrorPrint()
              }}
            >
              Print Error Report
            </Button>
          </div>
          <div style={{ display: 'none' }}>
            <div
              ref={contentRef}
              style={{
                margin: '50px',
                fontFamily: 'Arial Helvetica Sans-Serif',
                height: '100vh',
              }}
            >
              <h1 style={{ textAlign: 'center', marginBottom: '10px' }}>RSU Summary</h1>
              <br />
              <p>Below is the generated RSU summary report for all RSUs at {new Date().toISOString()} UTC:</p>
              <div style={{ marginTop: '25px' }}>
                <MaterialTable
                  columns={[
                    {
                      title: 'RSU',
                      field: 'rsu',
                      render: (rowData) => <p style={{ color: 'black', fontWeight: 'bold' }}>{rowData.rsu}</p>,
                    },
                    {
                      title: 'Road',
                      field: 'road',
                      render: (rowData) => <p style={{ color: 'black', fontWeight: 'bold' }}>{rowData.road}</p>,
                    },
                    {
                      title: 'Online Status',
                      field: 'online_status',
                      render: (rowData) => (
                        <p
                          style={
                            rowData.online_status.includes('RSU Offline')
                              ? {
                                  color: '#B60202',
                                  fontWeight: 'bold',
                                }
                              : {
                                  color: '#2B6510',
                                  fontWeight: 'bold',
                                }
                          }
                        >
                          {rowData.online_status}
                        </p>
                      ),
                    },
                    {
                      title: 'SCMS Status',
                      field: 'scms_status',
                      render: (rowData) => (
                        <p
                          style={
                            rowData.scms_status.includes('SCMS Healthy')
                              ? {
                                  color: '#2B6510',
                                  fontWeight: 'bold',
                                }
                              : {
                                  color: '#B60202',
                                  fontWeight: 'bold',
                                }
                          }
                        >
                          {rowData.scms_status}
                        </p>
                      ),
                    },
                  ]}
                  actions={[]}
                  data={props.rows}
                  title=""
                  options={{
                    toolbar: false,
                    search: false,
                    paging: false,
                  }}
                  style={{
                    backgroundColor: 'white',
                    color: 'black',
                    fontFamily: 'Arial Helvetica Sans-Serif',
                    border: 'none',
                  }}
                />
              </div>
            </div>
            <div
              ref={errorRef}
              style={{
                margin: '50px',
                fontFamily: 'Arial Helvetica Sans-Serif',
                height: '100vh',
              }}
            >
              <h1 style={{ textAlign: 'center', marginBottom: '10px' }}>RSU Error Summary</h1>
              <br />
              <p>Below is the generated RSU Error summary report for all RSUs at {new Date().toISOString()} UTC:</p>
              <div style={{ marginTop: '25px' }}>
                <MaterialTable
                  columns={[
                    {
                      title: 'RSU',
                      field: 'rsu',
                      render: (rowData) => <p style={{ color: 'black', fontWeight: 'bold' }}>{rowData.rsu}</p>,
                    },
                    {
                      title: 'Road',
                      field: 'road',
                      render: (rowData) => <p style={{ color: 'black', fontWeight: 'bold' }}>{rowData.road}</p>,
                    },
                    {
                      title: 'Online Status',
                      field: 'online_status',
                      render: (rowData) => (
                        <p
                          style={
                            rowData.online_status.includes('RSU Offline')
                              ? {
                                  color: '#B60202',
                                  fontWeight: 'bold',
                                }
                              : {
                                  color: '#2B6510',
                                  fontWeight: 'bold',
                                }
                          }
                        >
                          {rowData.online_status}
                        </p>
                      ),
                    },
                    {
                      title: 'SCMS Status',
                      field: 'scms_status',
                      render: (rowData) => (
                        <p
                          style={
                            rowData.scms_status.includes('SCMS Healthy')
                              ? {
                                  color: '#2B6510',
                                  fontWeight: 'bold',
                                }
                              : {
                                  color: '#B60202',
                                  fontWeight: 'bold',
                                }
                          }
                        >
                          {rowData.scms_status}
                        </p>
                      ),
                    },
                  ]}
                  actions={[]}
                  data={
                    props.rows !== undefined
                      ? props.rows.filter(
                          (row) =>
                            row.online_status.includes('RSU Offline') || row.scms_status.includes('SCMS Unhealthy')
                        )
                      : []
                  }
                  title=""
                  options={{
                    toolbar: false,
                    search: false,
                    paging: false,
                  }}
                  style={{
                    backgroundColor: 'white',
                    color: 'black',
                    fontFamily: 'Arial Helvetica Sans-Serif',
                    border: 'none',
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </ThemeProvider>
    </StyledEngineProvider>
  )
}

export default GenerateRSUErrorsPDF
