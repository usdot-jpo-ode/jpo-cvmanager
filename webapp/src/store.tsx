import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import intersectionReducer from './generalSlices/intersectionSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddIntersectionReducer from './features/adminAddIntersection/adminAddIntersectionSlice'
import adminAddUserReducer from './features/adminAddUser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditIntersectionReducer from './features/adminEditIntersection/adminEditIntersectionSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminOrganizationTabIntersectionReducer from './features/adminOrganizationTabIntersection/adminOrganizationTabIntersectionSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminIntersectionTabReducer from './features/adminIntersectionTab/adminIntersectionTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import adminNotificationTabReducer from './features/adminNotificationTab/adminNotificationTabSlice'
import adminAddNotificationReducer from './features/adminAddNotification/adminAddNotificationSlice'
import adminEditNotificationReducer from './features/adminEditNotification/adminEditNotificationSlice'
import menuReducer from './features/menu/menuSlice'
import asn1DecoderSlice from './features/intersections/decoder/asn1-decoder-slice'
import intersectionMapReducer from './features/intersections/map/map-slice'
import intersectionMapLayerStyleReducer from './features/intersections/map/map-layer-style-slice'
import dataSelectorReducer from './features/intersections/data-selector/dataSelectorSlice'
import mapSliceReducer from './pages/mapSlice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      intersection: intersectionReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddIntersection: adminAddIntersectionReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditIntersection: adminEditIntersectionReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminOrganizationTabIntersection: adminOrganizationTabIntersectionReducer,
      adminRsuTab: adminRsuTabReducer,
      adminIntersectionTab: adminIntersectionTabReducer,
      adminUserTab: adminUserTabReducer,
      adminNotificationTab: adminNotificationTabReducer,
      adminAddNotification: adminAddNotificationReducer,
      adminEditNotification: adminEditNotificationReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
      dataSelector: dataSelectorReducer,
      map: mapSliceReducer,
      asn1Decoder: asn1DecoderSlice,
    },
    preloadedState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        thunk: true,
        serializableCheck: false,
        immutableCheck: false,
      }),
    devTools: true,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
