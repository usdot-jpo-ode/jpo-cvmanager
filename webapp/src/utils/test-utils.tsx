/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container: HTMLElement) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val: string) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

import React from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

// Extend dayjs with timezone and UTC plugins
dayjs.extend(utc)
dayjs.extend(timezone)

// Mock component to set timezone
const MockLocalizationProvider = ({ children }) => {
  dayjs.tz.setDefault('America/Denver')
  return <LocalizationProvider dateAdapter={AdapterDayjs}>{children}</LocalizationProvider>
}

export { replaceChaoticIds, MockLocalizationProvider }
