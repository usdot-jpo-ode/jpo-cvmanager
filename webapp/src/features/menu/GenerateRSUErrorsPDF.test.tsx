import React from 'react'
import { render } from '@testing-library/react'
import GenerateRSUErrorsPDF from './GenerateRSUErrorsPDF'
import { Provider } from 'react-redux'
import { setupStore } from '../../store'
import { replaceChaoticIds } from '../../utils/test-utils'

jest.useFakeTimers().setSystemTime(new Date('2024-10-01'))

it('should take a snapshot', () => {
  const { container } = render(
    <Provider store={setupStore({})}>
      <GenerateRSUErrorsPDF />
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
