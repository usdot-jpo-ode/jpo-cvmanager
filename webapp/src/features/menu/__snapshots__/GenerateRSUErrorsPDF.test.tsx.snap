// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    class="containerDiv"
  >
    <div
      style="display: flex; justify-content: space-between; margin-bottom: 1rem;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
        style="color: white;"
        tabindex="0"
        type="button"
      >
        Print Full Report
        <span
          class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mocked-MuiButtonBase-root-MuiButton-root"
        style="color: white;"
        tabindex="0"
        type="button"
      >
        Print Error Report
        <span
          class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      style="display: none;"
    >
      <div
        style="margin: 50px; font-family: Arial Helvetica Sans-Serif; height: 100vh;"
      >
        <h1
          style="text-align: center; margin-bottom: 10px;"
        >
          RSU Summary
        </h1>
        <br />
        <p>
          Below is the generated RSU summary report for all RSUs at 
          2024-10-01T00:00:00.000Z
           UTC:
        </p>
        <div
          style="margin-top: 25px;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-mocked-MuiPaper-root"
            style="position: relative; background-color: white; color: black; font-family: Arial Helvetica Sans-Serif;"
          >
            <div
              class="MuiBox-root css-187halg"
            >
              <div>
                <div
                  style="overflow-y: auto;"
                >
                  <div>
                    <table
                      class="MuiTable-root css-mocked-MuiTable-root"
                      style="table-layout: auto;"
                    >
                      <thead
                        class="MuiTableHead-root css-mocked-MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head css-mocked-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="0"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="0"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                RSU
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="1"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="1"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Road
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="2"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="2"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Online Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-0-hidden-text-0"
                              data-rfd-drag-handle-context-id="0"
                              data-rfd-drag-handle-draggable-id="3"
                              data-rfd-draggable-context-id="0"
                              data-rfd-draggable-id="3"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                SCMS Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root css-mocked-MuiTableBody-root"
                      >
                        <tr
                          class="MuiTableRow-root css-mocked-MuiTableRow-root"
                          style="height: 49px;"
                        >
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
                            colspan="4"
                            style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                          >
                            No records to display
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin: 50px; font-family: Arial Helvetica Sans-Serif; height: 100vh;"
      >
        <h1
          style="text-align: center; margin-bottom: 10px;"
        >
          RSU Error Summary
        </h1>
        <br />
        <p>
          Below is the generated RSU Error summary report for all RSUs at 
          2024-10-01T00:00:00.000Z
           UTC:
        </p>
        <div
          style="margin-top: 25px;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-mocked-MuiPaper-root"
            style="position: relative; background-color: white; color: black; font-family: Arial Helvetica Sans-Serif;"
          >
            <div
              class="MuiBox-root css-187halg"
            >
              <div>
                <div
                  style="overflow-y: auto;"
                >
                  <div>
                    <table
                      class="MuiTable-root css-mocked-MuiTable-root"
                      style="table-layout: auto;"
                    >
                      <thead
                        class="MuiTableHead-root css-mocked-MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head css-mocked-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="0"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="0"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                RSU
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="1"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="1"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Road
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="2"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="2"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                Online Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-mocked-MuiTableCell-root"
                            scope="col"
                            style="box-sizing: border-box;"
                          >
                            <div
                              aria-describedby="rfd-hidden-text-1-hidden-text-6"
                              data-rfd-drag-handle-context-id="1"
                              data-rfd-drag-handle-draggable-id="3"
                              data-rfd-draggable-context-id="1"
                              data-rfd-draggable-id="3"
                              draggable="false"
                              role="button"
                              style="position: relative; min-width: 0; display: flex;"
                              tabindex="0"
                            >
                              <span
                                aria-sort="none"
                                class="MuiButtonBase-root MuiTableSortLabel-root css-mocked-MuiButtonBase-root-MuiTableSortLabel-root"
                                data-testid="mtableheader-sortlabel"
                                role="columnheader"
                                tabindex="0"
                              >
                                SCMS Status
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-mocked-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                  data-testid="arrow_downward"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8z"
                                  />
                                </svg>
                              </span>
                              <span
                                data-testid="material-table-column-sort-order-indicator"
                                style="width: 1em;"
                              />
                            </div>
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root css-mocked-MuiTableBody-root"
                      >
                        <tr
                          class="MuiTableRow-root css-mocked-MuiTableRow-root"
                          style="height: 49px;"
                        >
                          <td
                            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-mocked-MuiTableCell-root"
                            colspan="4"
                            style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                          >
                            No records to display
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
